### DISCUSSION TITLE
Feature request: detect ordinal domain in color when ordering by same channel

### DISCUSSION THREAD

**Discussion by jimjam-slam** [observablehq/plot#2001] [Q&A] [answered]
(Possibly a duplicate of #1121, but I'm not 100% sure!)

When I make a barchart in which the stacking channel (`fill` or `color`) is ordinal, it can be tricky to get the legend and the stacking order to line up. Although there are many cases where simple alphanumerical order or frequency is the right call, often (especially with survey responses) it's easier to just list out the levels.

I had assumed that if I provided an array of values to `color.domain` (which orders the legend glyphs), the stacking order would reflect this.

I imagine there are cases where you do still want to stack by frequency even when your legend is ordered (and `sort` can ), but I think if you provide the same channel to both `fill`/`color` and `sort`, users would benefit from Plot using any levels provided by `color.domain`. The workaround is to provide a sorting function that re-uses those levels, but I think this is a common enough scenario that it would be helpful for Plot to default to this behaviour.

Here's an example notebook:

https://observablehq.com/@jimjamslam/stack-a-barchart-on-ordinal-data

Thank you!

**Comment by Fil** [comment, marked as answer]
Congratulations on hitting #2000 :)

The solution is to use `order: freqLevels`. ("If an array of z values are specified, they should enumerate the z values for all series in the desired order")

**Reply by jimjam-slam** [reply]
Thanks very much! Using `order: freqLevels` works perfectly 😁

I'd just started to see the term "z values" pop up a few times in the docs for help on this. ~~Do "z values" refer to values along any channel that isn't `x` or `y` (or maybe also `fx` or `fy`)? Or does it have a more specific meaning?~~

~~I don't think I've seen a definition of "z values" in the docs, but I do see it used in the [group transform](https://observablehq.com/plot/transforms/group) docs, suggesting that "z values" means "values for the grouping channel".~~

I probably should have realised that the stack transform is grouping the data under the hood—it does say it right there on the page several times—but I think this sentence from the group docs would be really useful to include in the stack docs: "You can partition groups using z. If z is undefined, it defaults to fill or stroke, if any." Or maybe even just finishing the bullet list with, "An array of z ([grouping](https://observablehq.com/plot/transforms/group)) values"!
